src/main/scala/cosas/records/syntax.scala

```diff
@@ -4,37 +4,42 @@ import ohnosequences.cosas._, types._, fns._, klists._
 
 case object syntax {
 
-  final case class RecordTypeDenotationSyntax[RT <: AnyRecordType, Vs <: RT#Keys#Raw](val vs: Vs) extends AnyVal {
+  final case class RecordTypeDenotationSyntax[RT <: AnyRecordType, Vs <: RT#Raw](val vs: Vs) extends AnyVal {
 
-    def get[D <: AnyDenotation { type Tpe = T }, T <: AnyType](tpe: T)(implicit
-      get: AnyApp1At[D findIn Vs, Vs] { type Y = D }
-    ): D =
-      get(vs)
-
-    def altGet[D <: AnyDenotation { type Tpe = T }, T <: AnyType, O <: AnyKList](tpe: T)(implicit
-      p: AnyApp1At[PickS[AnyDenotation { type Tpe = T }], Vs] { type Y = (D,O) }
+    def get[D <: AnyDenotationOf[T], T <: AnyType](tpe: T)(implicit
+      p: AnyApp1At[FindS[AnyDenotationOf[T]], Vs] { type Y = D }
     )
     : D =
-      p(vs)._1
+      p(vs)
 
-      def altGet2[D <: AnyDenotation { type Tpe = T }, T <: AnyType](tpe: T)(implicit
-        p: AnyApp1At[FindS[AnyDenotation { type Tpe = T }], Vs] { type Y = D }
-      )
-      : D =
-        p(vs)
+    def getV[D <: AnyDenotationOf[T], T <: AnyType](tpe: T)(implicit
+      p: AnyApp1At[FindS[AnyDenotationOf[T]], Vs] { type Y = D }
+    )
+    : D#Value =
+      p(vs).value
 
-    def getV[D <: AnyDenotation { type Tpe = T }, T <: AnyType](tpe: T)(implicit
-      get: AnyApp1At[D findIn Vs, Vs] { type Y = D }
-    ): D#Value =
-      get(vs).value
+    def update[S <: AnyKList.withBound[AnyDenotation]](s: S)(implicit
+      repl: AnyApp2At[replace[Vs], Vs, S] { type Y = Vs }
+    )
+    : RT := Vs =
+      new (RT := Vs)(repl(vs,s))
 
   }
 
   final case class RecordTypeSyntax[RT <: AnyRecordType](val rt: RT) extends AnyVal {
 
-    def apply[Vs <: AnyKList { type Bound = AnyDenotation }](values: Vs)(implicit
-      reorder: AnyApp1At[TakeFirst[RT#Raw], Vs]
-    ): RT := RT#Raw = rt := reorder(values)
+    def apply[Vs <: AnyKList.withBound[AnyDenotation], RTRaw <: RT#Raw](values: Vs)(implicit
```

Comment:
If I understood https://github.com/ohnosequences/cosas/pull/76#issuecomment-154698086 right, you suggest to rename it to `reorder`. Fine for me.